### ISSUE TITLE
1.7 RC 3 issue with attachments

### ISSUE THREAD

**Issue by ghost** [osTicket/osTicket-1.7#347]
If you have attachments enabled  and someone opens a ticket when i view the ticket it  will alow the attchment to go thru and everythings seems find but  everytime they reply back and forth it will say system message

11/02/2012 1:36 am  File Upload Error   SYSTEM
Invalid file type for 

so even if they dont attach a file,   it will display this when i am logged in as admin  , it wont display this message on their side...

also still people are only able to upload 1 attachment even tho i have selected they can upload 10

**Comment by MikeVeg** [comment]
I got the same problems. 

I can only upload one file in IE - works fine in all other browsers (what a surprise!)

I also see the "Invalid file type" in Ticket Thread, but I can live with that. My users need to upload at least two files so that is a big problem for me.

**Comment by ghost** [comment]
Thats odd,   i can only upload 1 file in Firefox, Opera, google chrome

but in internet explorer i click the browse button to locate the file then  it doesnt acually upload the file at all

all the other browsers i upload the first file and there is not another button were i can click browse to upload a 2nd file

**Comment by MikeVeg** [comment]
Strange... try and visit this page (http://fusionbox.org/projects/jquery-multifile-js/) and see if jquery-multifile works with different browsers. It is the original script and may help narrow down the problem.

**Comment by ghost** [comment]
Yeah that is strange,   i visited the link you provided and the multi file worked perfectly,   the button that says add file or choose file depending on browser remains their while i am attaching multiple files

But for some reason on my host after i click on the button once and upload the file,  the button goes away so it can not upload any more...

Just like you i need a minimum of 2,   I am not sure if it has something to do with the host...

I am using Linux php 5.2  most of the settings i left at default except i changed, enable FAQ  and enabled web upload and email upload..

Are you running your version on a localhost

**Comment by MikeVeg** [comment]
Does it also work with IE on the fusionbox.org page? I doesn't work on my IE9... I can't upload anything no matter what I do. I also used another computer with IE9, but same problem.

I tried using IE8 on a laptop (with the above site), but then I get a warning (yellow bar with a certificate error). I can upload files with IE8, but only if I choose to ignore the warning and show blocked content.

Yes, I am running my version on a localhost using WampServer (php 5.3), but I don't think that has anything to do with it since I got problems with the script on the fusionbox.org site as well.

**Comment by ghost** [comment]
Yeah i just tried http://fusionbox.org/  with ie and it doesnt work with  i e at all but it works with all the other browsers...

also i checked they are using the same  host as i am  http://fusionbox.org/ is using godaddy the only thing i can possibly think of is  http://fusionbox.org/  is only compatible with  php 5.3 and not 5.2

I havent switched my live server at godaddy to 5.3  yet cause somethings still require 5.2  but if i am a betting person the reason why its not working on my site in any browser is because its only supported by 5.3

But what i did, is contacted the developer of the jquery multi file,   and asked them if its only supported by php 5.3  this will answer my question...

then i informed them that it does not work on their site in internet explorer 9 but works in all other browsers,  so i guess i will wait and see what they say,  and if he responds ill let ya know what he says about ie9

**Comment by protich** [comment]
@jason102178  - We'll look into the issue - hosting can't be the cause at all. Make sure you've actually enabled multiple file support in Admin Panel > Settings > Tickets, otherwise it will be limited to one file by default.

Please note that I heavily modified/customized the original version developed by Rocky @fusionbox. It's no longer same or comparable scripts, they can't help you with osTicket related issues. I've had them contact me because _you_ contacted them for help. If you want to switch to what they're using them feel free to do so, but note that it won't work with osTicket settings.

**Comment by MikeVeg** [comment]
@jason102178 - I forgot to say that I contacted the developer of the jquery multi file yesterday... he said that I could open an issue on  https://github.com/fusionbox/jquery-multifile, and he also linked to this issue. I thought that this osTicket version was the most relevant one because I use that system (and I saw your comment), but I will try to open an issue for the fusionbox plugin as well.

It doesn't work on my local testsite, using IE and wamp with php 5.3, so I don't think that is the problem.

**Comment by protich** [comment]
@MikeVeg - No problem. We'll try to track down the issue. I just wanted to make sure we're all on the same page in regard to both plugins.

**Comment by MikeVeg** [comment]
@protich - Thanks - that is great, I really need this to work (I was writing my last comment at the same time as you did...) Multiple file support in Admin Panel is on here (but not the place you said - I'm using v1.7 - RC2), and I can upload multiple files in every other browser than IE.

I tried to turn on script error warnings in IE9, and I got this error: "Object does not support property or method 'cloneNode', jquery-1.7.2.min.js, line 4" - not sure if that helps you...

**Comment by protich** [comment]
@MikeVeg - Thank you very much. The error/warning is very helpful - it gives us something to work with. 

PS. You might want to upgrade to RC3 - comes with bug fixes and a few enhancements.

**Comment by MikeVeg** [comment]
@protich - Thanks, I will for sure upgrade to RC3. Please let me know if there is anything I can assist with - the original version developed by Rocky @fusionbox gave me some other errors/warnings, but they are perhaps not relevant (IE was also the problem here but maybe that doesn't say much... damn IE)

**Comment by ghost** [comment]
@protich  I have  allow attachment(global setting) and allow web upload    enabled and  user and staf max upload set for 5 files  this is all the settings that should be enabled to make it work right?

I havent seen no settings per say that says enable mutiple file support,   and i didnt know you have heavily modified it,  but no i would never tried to change the way the plugin works.

**Comment by protich** [comment]
@jason102178  - Yes, that is all you have to do. Setting max upload greater than 1 basically enables it. Are you getting any errors?

**Comment by ghost** [comment]
This is the only error were getting in the admin

http://i4.photobucket.com/albums/y101/jason102178/new/error.png

Heres a couple of images of how it looks on front end
1. Before i add attachment looks like this
   http://i4.photobucket.com/albums/y101/jason102178/new/beforeattachment.png

2.After i add attachment looks like this
http://i4.photobucket.com/albums/y101/jason102178/new/afterattachment.png

I am not getting any kind of error on the user side

**Comment by protich** [comment]
- We'll resolve the "Invalid file type" issue
- #2 is correct but another input box should be showing up. Is this happening across all browsers or just IE?

**Comment by ghost** [comment]
Its happening accross all browsers,    and  IE doesnt let you upload anything

**Comment by ghost** [comment]
I dont know if providing you with the test osticket 1.7 rc3 installation would help any but if you would like it i could post it.

**Comment by MikeVeg** [comment]
Sorry to ask, but any idea on how long it will take to fix this? If it is too complex would it then be possible to insert something like this in the meantime (I'm not allowed to paste complete code here so there are no < and >):

!--[if IE]
input type="button" value="Upload Files" style="float: left;" onclick="$('#attachments').click();"/
input type="file" class="multifile" multiple="multiple" name="attachments[]" id="attachments" size="8" value="" style="visibility:hidden"/
![endif]

This way you get one "Upload Files" button, and you get rid of the empty path field (or whatever the name is) that seems to confuse a lot of people.

I'm no great coder, and with the code above you can only press the button a couple of times before it doesn't work anymore... but perhaps it is easy to clear the value each time you press the button? Then it would work... or at least it would work for me.

**Comment by klimnad** [comment]
for me do not work attachments when i allow send attachments to client by  email - letter  recieved by client is non readable with binary content

**Comment by ghost** [comment]
@MikeVeg  the problem has been fixed now on the fusionbox.org,  i spoke with the developer rocky and he said
There was an issue with IE9 and using the version from github raw pages. It should send the correct mime type,  so he released the fix on github for this now.

**Comment by MikeVeg** [comment]
@jason102178 thanks, I also got an email from him. As he said, the problem with osTicket's script is probably not related to the problem there was with the original script since the osTicket JS isn't being hosted on GitHub.

It would be nice to know if this issue will take days, weeks, months to fix - it would be sad if I had to find an alternative solution after spending weeks on learning and customizing osTicket.

**Comment by protich** [comment]
@MikeVeg - I haven't had time to debug the issue (hint: I don't do osTicket full-time), but I'll be revisiting the issue in the next couple of days.

> osTicket JS isn't being hosted on GitHub

osTicket's entire [code base](https://github.com/osTicket/osTicket-1.7) is on github including the [said script](https://github.com/osTicket/osTicket-1.7/blob/develop/js/jquery.multifile.js). It's a little ironic since we're having the discussion on GitHub.

Thanks for the patience.

**Comment by MikeVeg** [comment]
@protich I appreciate and I fully understand that you have other things in your life than osTicket.
osTicket is great in so many ways, but the ability to upload files is a pretty important feature.

BTW: maybe I used the wrong words, but what I meant about "osTicket JS isn't being hosted on GitHub" is that the problem with the original script on fusionbox.org was caused by using GitHub's raw file feature as the host for the JavaScript file, and it sends the wrong Content Type (as explained by the developer from fusionbox). 

Thanks for your time.

**Comment by star3132** [comment]
Just two cents, I was having the issue above with Internet Explorer not allowing multiple file uploads.  I edited the /js/jquery.multifile.js file and commented out the following lines: (starts on mine at line 153)

``` javascript
  //Clone file input and clear the value without triggering a warning!
  $.fn.multifile.cloneInput = function(input) {

      var $clone = input.clone(true);

//      if ($.browser.msie) {
//          $clone.replaceWith(function () { return $(this).clone(true); });
//      } else {
          $clone.val('');
//      }

      return $clone;
  }
```

This allowed IE to upload multiple documents.

Again, worked for me.

**Comment by MikeVeg** [comment]
Thanks star3132. Do you have any idea what those lines of code does? Are they important in any other way - in regard to security etc.?

**Comment by star3132** [comment]
It appears to be a workaround for an older version of IE that perhaps no longer applies.  I don't think it has any security concerns, but i can't be 100% sure.  Thanks.
